### PR TITLE
replace `request` with `fetch`

### PR DESCRIPTION
**Pull request by MichaelAlexanderBanuelos**
# Description

replacing request with fetch

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

**Test Configuration**:

- Firmware version:
- Hardware:
- Toolchain:
- SDK:

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by MichaelAlexanderBanuelos** [comment]
@MustafaAliDS we'll need to uninstall `request` and its types. you can do so with `npm uninstall request @types/request`


i can do that tomorrow or you can batch it in with some other work or pull this branch down and add that commit

**Comment by MustafaAliDS** [comment]
Reason why I removed:

```
   const {
        data: { body },
      } = spotifyResponse;

      const ACCESS_TOKEN = body.access_token;
      const TOKEN_TYPE = body.token_type;
      const SCOPE = body.scope;
      const EXPIRES_IN = body.expires_in;

      res.send({
        access_token: ACCESS_TOKEN,
        token_type: TOKEN_TYPE,
        scope: SCOPE,
        expires_in: EXPIRES_IN,
      });
```

is because we would get hit with an empty object error by the catch block. The code I changed will send us back the desired JSON w/ the access_token and other info in the browser, which didn't happen before.

File: src/routes/login.ts [hunk 1]
```diff
@@ -1,10 +1,10 @@
 import express, { Request, Response, RequestHandler } from 'express';
-
 import querystring from 'querystring';
 
 const loginRouter = express.Router();
 
 import * as dotenv from 'dotenv';
+import { Body } from 'tsoa';
```
Comment (MichaelAlexanderBanuelos) [inline src/routes/login.ts:7]: we shouldn't need to touch anything from `tsoa` yet

**Comment by MichaelAlexanderBanuelos** [comment]
weird... the github action workflow didn't run on the latest commit. i'd expect it to run and fail due to a naming collisoin from `Body`

**Pull request merged**